From 8d1c0fc76384b9361e71564c121a6c7da85ce42a Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Fri, 16 Oct 2026 02:12:50 +0000
Subject: [PATCH] always use speed mode ahead-of-time compilation

Port of GrapheneOS 3156bd84 + c09e1c36 to the Android 15 property list: with the JIT disabled, compile apps fully ahead of time; keep speed-profile after OTAs so updates don't take too long.
---
 target/product/runtime_libart.mk | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/target/product/runtime_libart.mk b/target/product/runtime_libart.mk
index 0ff38f7..767c28a 100644
--- a/target/product/runtime_libart.mk
+++ b/target/product/runtime_libart.mk
@@ -110,20 +110,20 @@ PRODUCT_SYSTEM_PROPERTIES += \
 # without exceptions).
 # TODO(b/243646876): Remove `pm.dexopt.post-boot`.
 PRODUCT_SYSTEM_PROPERTIES += \
-    pm.dexopt.post-boot?=verify \
-    pm.dexopt.first-boot?=verify \
-    pm.dexopt.boot-after-ota?=verify \
-    pm.dexopt.boot-after-mainline-update?=verify \
-    pm.dexopt.install?=speed-profile \
-    pm.dexopt.install-fast?=skip \
-    pm.dexopt.install-bulk?=speed-profile \
-    pm.dexopt.install-bulk-secondary?=verify \
-    pm.dexopt.install-bulk-downgraded?=verify \
-    pm.dexopt.install-bulk-secondary-downgraded?=verify \
-    pm.dexopt.bg-dexopt?=speed-profile \
-    pm.dexopt.ab-ota?=speed-profile \
-    pm.dexopt.inactive?=verify \
-    pm.dexopt.cmdline?=verify \
+    pm.dexopt.post-boot?=speed \
+    pm.dexopt.first-boot?=speed \
+    pm.dexopt.boot-after-ota?=speed-profile \
+    pm.dexopt.boot-after-mainline-update?=speed \
+    pm.dexopt.install?=speed \
+    pm.dexopt.install-fast?=speed \
+    pm.dexopt.install-bulk?=speed \
+    pm.dexopt.install-bulk-secondary?=speed \
+    pm.dexopt.install-bulk-downgraded?=speed \
+    pm.dexopt.install-bulk-secondary-downgraded?=speed \
+    pm.dexopt.bg-dexopt?=speed \
+    pm.dexopt.ab-ota?=speed \
+    pm.dexopt.inactive?=speed \
+    pm.dexopt.cmdline?=speed \
     pm.dexopt.shared?=speed
 
 ifneq (,$(filter eng,$(TARGET_BUILD_VARIANT)))
-- 
2.43.0

